fury/text_tools.py

```diff
@@ -0,0 +1,589 @@
+            # a = face.get_advance(gindex, FT_LOAD_RENDER |
+            # FT_LOAD_TARGET_LCD)/(64*72)
+            # glyph.advance = a, glyph.advance[1]
+        self._calc_relative_sizes()
+
+    def _calc_relative_sizes(self):
+        for char, glyph in self.glyphs.items():
+            glyph.relative_size = np.array(
+                [glyph.w, glyph.h])/self.max_glyphy_size
+            glyph.relative_offset = np.array(glyph.offset)/self.size
+            glyph.hmax = self.max_glyphy_size[1]
+            if glyph.h > 0:
+                glyph.pad = (glyph.stb)/glyph.h
+
+
+class TextureGlyph:
+    """
+    A texture glyph gathers information relative to the size/offset/advance and
+    texture coordinates of a single character. It is generally built
+    automatically by a TextureFont.
+
+    """
+
+    def __init__(self, charcode, size, offset, advance, texcoords, px):
+        """
+        Build a new texture glyph
+
+        Parameter:
+        ----------
+
+        charcode : char
+            Represented character
+
+        size: tuple of 2 ints
+            Glyph size in pixels
+
+        offset: tuple of 2 floats
+            Glyph offset relatively to anchor point
+
+        advance: tuple of 2 floats
+            Glyph advance
+
+        texcoords: tuple of 4 floats
+            Texture coordinates of bottom-left and top-right corner
+            of glyph bounding box
+
+        """
+        self.charcode = charcode
+        self.size = size
+        self.relative_size = np.array([1., 1.])
+        self.px = px
+        self.offset = offset
+        self.advance = advance
+        self.texcoords = texcoords
+        self.kerning = {}
+
+    def get_kerning(self, charcode):
+        """ Get kerning information
+
+        Parameters:
+        -----------
+
+        charcode: char
+            Character preceding this glyph
+        """
+        if charcode in self.kerning.keys():
+            return self.kerning[charcode]
+        else:
+            return 0
+
+
+def create_bitmap_font(
+        font_size=7, font_path=None,
+        atlas_size=(1024, 1024),
+        show=False, save_path=None):
+    """This function is used to create a bitmap font.
+
+    Parameters
+    ----------
+    font_size : int
+        The size of the font.
+    font_path : str
+        The path to the font file.
+    pad : int
+        The padding of the font.
+    atlas_size : tuple, optional
+        The size of the texture atlas in pixels.
+    show : bool
+        Whether to show the result.
+    save_path : str
+        The path to save the image file.
+
+    Returns
+    -------
+    image_array : ndarray
+        The image array.
+    char2pos : dict
+        A dictionary that maps characters to their positions in the
+        numpy array.
+
+    """
+
+    if font_size == 7 and font_path is None:
+        font_path = f'{fury.__path__[0]}/data/files/FreeMono'
+        image_arr = Image.open(font_path+'.bmp')
```

Comment:
Why 7?  Is this going to be bundled with fury? In that case there is another way to provide data. Checkout the network example that shows a wiki network. How about generating this in the building stage? As if there are changes in future, we the atlas will be always updated.